src/app/pages/catalog/catalog.component.ts

```diff
@@ -14,9 +14,18 @@ import { AsyncPipe } from '@angular/common';
 })
 export class CatalogComponent implements OnInit {
   private readonly categoriesService = inject(CategoriesService);
-  categories$!: Observable<QuizCategory[]>;
+  private readonly categoriesStoreService = inject(CategoriesStoreService);
+
+  categories$!: Observable<any>;
 
   ngOnInit(): void {
-    this.categories$ = this.categoriesService.getRandomCategories();
```

Comment:
how to  optimise this code:

  ngOnInit(): void {
    this.categoriesService.getRandomCategories().subscribe();
     
    this.categories$ = this.categoriesStoreService.getCategories()
}
    
In categories service:

getRandomCategories(): Observable<QuizCategory[]> { 
    if(categories && categories.length > 0) {
        return this.categoriesStoreService.getCategories()
    }
    
    return this.apiService.fetchCategories().pipe(
      map(categories => this.randomizationService
      .getRandomItems(categories, this.categoriesNumber)
      .map(category => this.enrichCategory(category))),
            tap((categories) => {
        this.categoriesStoreService.addCategories(categories);
      }),
    );
}